public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/url_builder.ts

```diff
@@ -11,9 +11,16 @@ export default class UrlBuilder {
     const resourceNameArray = resourceName.split('/');
     const provider = metricDefinitionArray.shift();
     const urlArray = ['/subscriptions', subscriptionId, 'resourceGroups', resourceGroup, 'providers', provider];
-    for (const i in metricDefinitionArray) {
-      urlArray.push(metricDefinitionArray[i]);
-      urlArray.push(resourceNameArray[i]);
+    if (metricDefinition.startsWith('Microsoft.Storage/storageAccounts/') && resourceNameArray.at(-1) !== 'default') {
+      resourceNameArray.push('default');
+    }
```

Comment:
The only issue I can see with this is that (afaik) only the storage account namespace requires `default` to be appended. This could cause a problem if there are other namespaces with a similar structure that don't require the additional suffix.

Reply:
Otherwise I think you've covered all the possible cases!

Reply:
mmh, all this one-by-one cases are super brittle. Let's go ahead with this but we should re-evaluate this, we should remove hard-coded values and use API information for these cases, it's making the UI weird and will cause more bugs in the future for cases we are not aware of.

Reply:
I've just thought of a slight variation of case 2:

`metricDefinition = $ns`
`resourceName = $rs`
`$ns = 'ms.Storage/storageAccounts/tableServices'`
`$rs = 'foo`

This will have to be handled the same as case 2 I believe as we still will not be able to appropriately construct a resource URI - thoughts @andresmgot?

Reply:
yes, the same for:

`metricDefinition = $ns/tableServices`
`resourceName = $rs`
`$ns = 'ms.Storage/storageAccounts'`
`$rs = 'foo/bar'`

(in this case, it's the resource name is the one problematic)